fe/fe-core/src/main/java/com/starrocks/catalog/CatalogRecycleBin.java

```diff
@@ -253,7 +293,7 @@ private synchronized void eraseDatabaseWithSameName(String dbName) {
             Database db = dbInfo.getDb();
             if (db.getFullName().equals(dbName)) {
                 iterator.remove();
-                idToRecycleTime.remove(entry.getKey());
+                removeRecycleMarkers(entry.getKey());
```

Comment:
Your solution can not cover this case: Drop a table and create a same name table, the dropped table must be erased immediately, even thought it has been marked as erase later.